2023-03-29 16:30:25,974 : INFO : root : Loading config.
2023-03-29 16:30:26,006 : INFO : YamlMultifileConfigLoader : Machine config file #1: config.yaml
2023-03-29 16:30:26,006 : INFO : ConfigProcessor : Loading config from cache: C:\Users\digidev\AppData\Local\Temp\207a4012de915441e2519cedb288d8b5.mpf_cache
2023-03-29 16:30:27,158 : INFO : ConfigProcessor : Loading config from cache: C:\Users\digidev\AppData\Local\Temp\ae663e8f24481bd52f7f5e9610d87d3b.mpf_cache
2023-03-29 16:30:27,158 : INFO : ConfigProcessor : Loading config from cache: C:\Users\digidev\AppData\Local\Temp\d7dbd5ece0283f90787e254e6a383d33.mpf_cache
2023-03-29 16:30:27,158 : INFO : Machine : Mission Pinball Framework Core Engine v0.56.0
2023-03-29 16:30:27,158 : INFO : Machine : Command line arguments: {'no_load_cache': False, 'create_config_cache': True, 'bcp': True, 'configfile': ['config.yaml'], 'force_assets_load': False, 'jsonlogging': False, 'logfile': 'logs\\2023-03-29-16-30-25-mpf-DESKTOP-JGO3PQG.log', 'pause': False, 'production': False, 'text_ui': True, 'loglevel': 15, 'consoleloglevel': 20, 'force_platform': None, 'syslog_address': None, 'mc_file_name': None, 'no_sound': False}
2023-03-29 16:30:27,158 : INFO : Machine : MPF path: C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf
2023-03-29 16:30:27,158 : INFO : Machine : Machine path: C:\Users\digidev\source\Pinball_Three-Amigos
2023-03-29 16:30:27,158 : INFO : Machine : Platform: win32
2023-03-29 16:30:27,158 : INFO : Machine : Python executable location: C:\Users\digidev\.local\pipx\venvs\mpf\Scripts\python.exe
2023-03-29 16:30:27,158 : INFO : Machine : Python version: 3.9.13 (64-bit)
2023-03-29 16:30:27,252 : INFO : Machine : Initialise MPF.
2023-03-29 16:30:27,512 : ERROR : Machine : Failed to initialise MPF
Traceback (most recent call last):
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\platforms\p_roc_common.py", line 31, in <module>
    import pinproc
ModuleNotFoundError: No module named 'pinproc'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\platforms\p_roc_common.py", line 39, in <module>
    from mpf.platforms.pinproc.windows import pinproc
ImportError: cannot import name 'pinproc' from 'mpf.platforms.pinproc.windows' (unknown location)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\core\machine.py", line 679, in initialise_mpf
    raise init.exception()
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\core\machine.py", line 246, in initialise
    await self.initialise_core_and_hardware()
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\core\machine.py", line 232, in initialise_core_and_hardware
    self._load_hardware_platforms()
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\core\machine.py", line 479, in _load_hardware_platforms
    self.add_platform(hardware_platform)
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\core\machine.py", line 613, in add_platform
    self.hardware_platforms[name] = hardware_platform(self)
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\platforms\p3_roc.py", line 113, in __init__
    super().__init__(machine)
  File "C:\Users\digidev\.local\pipx\venvs\mpf\lib\site-packages\mpf\platforms\p_roc_common.py", line 180, in __init__
    raise MpfRuntimeError('Could not import "pinproc". Either the library is not installed or is missing '
mpf.exceptions.runtime_error.MpfRuntimeError: Runtime Error in P-Roc: Could not import "pinproc". Either the library is not installed or is missing some of its dependencies. Check the install instructions for your OS in Multimorphic section of the MPF docs. You can run mpf with "-X" to use virtual hardware in the meantime. Error Code: RE-P-Roc-3 (https://docs.missionpinball.org/en/0.56/logs/RE-P-Roc-3.html)
2023-03-29 16:30:31,032 : INFO : root : MPF run loop ended.
